eslint.config.mjs

```diff
@@ -0,0 +1,436 @@
+import globals from "globals";
+
+import import_ from "eslint-plugin-import";
+import jasmine from "eslint-plugin-jasmine";
+import json from "eslint-plugin-json";
+import noUnsanitized from "eslint-plugin-no-unsanitized";
+import perfectionist from "eslint-plugin-perfectionist";
+import prettierRecommended from "eslint-plugin-prettier/recommended";
+import unicorn from "eslint-plugin-unicorn";
+
+const jsFiles = folder => {
+  const prefix = folder === "." ? "" : folder + "/";
+  return [prefix + "**/*.js", prefix + "**/*.jsm", prefix + "**/*.mjs"];
+};
+
+// Include all files referenced in extensions/chromium/background.js
+const chromiumExtensionServiceWorkerFiles = [
+  "extensions/chromium/options/migration.js",
+  "extensions/chromium/preserve-referer.js",
+  "extensions/chromium/pdfHandler.js",
+  "extensions/chromium/extension-router.js",
+  "extensions/chromium/suppress-update.js",
+  "extensions/chromium/telemetry.js",
+];
+
+export default [
+  {
+    ignores: [
+      "**/build/",
+      "**/l10n/",
+      "**/docs/",
+      "**/node_modules/",
+      "external/bcmaps/",
+      "external/builder/fixtures/",
+      "external/builder/fixtures_babel/",
+      "external/quickjs/",
+      "external/openjpeg/",
+      "test/tmp/",
+      "test/pdfs/",
+      "web/locale/",
+      "**/*~/",
+    ],
+  },
+
+  /* ======================================================================== *\
+                             Base configuration
+  \* ======================================================================== */
+
+  prettierRecommended,
+  {
+    files: ["**/*.json"],
+    ...json.configs.recommended,
+  },
+  {
+    files: jsFiles("."),
+    ignores: chromiumExtensionServiceWorkerFiles,
+    languageOptions: {
+      globals: globals.browser,
+    },
+  },
+  {
+    files: jsFiles("."),
+
+    plugins: {
+      import: import_.flatConfigs.recommended.plugins.import,
```

Comment:
*Perhaps this is a stupid question, based on a lack of understand of the flat-config format:*

Currently we're not using any default-config for the `import` plugin, but explicitly specify the rules we want. Does this line mean that we now *potentially* enable rules that we previously were not using?

(If so that might be fine, I simply want us to avoid any "surprises" later on.)

Reply:
No! This is a bit of a hack, but `eslint-plugin-import` doesn't properly expose a way of using the plugin with the flat config unless you use their recommended config.

To use their recommended config, you'd add `import_.flatConfigs.recommended,` to the `export default` array, similar to how it's done for prettier in this PR.

However, in this line I'm grabbing _just the plugin_ from their recommended config, which seems to be the only way to do it, and then defining our own list of rules.